src/csharp/Microsoft.Spark/Interop/Ipc/JvmThreadPoolGC.cs

```diff
@@ -97,16 +98,17 @@ public bool TryAddThread(Thread thread)
         /// corresponding JVM thread will also be disposed.
         /// </summary>
         /// <param name="threadId">The ID of the thread to remove.</param>
+        /// <param name="pid">The Id of the current process.</param>
         /// <returns>True if success, false if the thread cannot be found.</returns>
-        private bool TryDisposeJvmThread(int threadId)
+        private bool TryDisposeJvmThread(int threadId, int pid)
         {
             if (_activeThreads.TryRemove(threadId, out _))
             {
                 // _activeThreads does not have ownership of the threads on the .NET side. This 
                 // class does not need to call Join() on the .NET Thread. However, this class is
                 // responsible for sending the rmThread command to the JVM to trigger disposal
                 // of the corresponding JVM thread.
-                if ((bool)_jvmBridge.CallStaticJavaMethod("DotnetHandler", "rmThread", threadId))
+                if ((bool)_jvmBridge.CallStaticJavaMethod("DotnetHandler", "rmThread", pid + "_" + threadId))
```

Comment:
Shall we just pass process id separately?

Reply:
hey @imback82 , sure, **sperate process id to pass the dotnetbackend** is also one way to fix the issue.

My understand is that current implement is also ok, right? for fix current issue, seperate process id seems like not bring more benefits while needs more code change?

Or do you have any strong idea to share to use seperate process id? :)

Reply:
Yea, the current approach is fine. But, having a separate process id seems better for the maintainability reason. I already see three different places (including test) where concatenation logic is introduced. Also, on the server side, if we want to log the info, we need to have a logic to separate process and thread id for a better message. (I also suggest using a tuple as a key (process id, thread id) to make it clear; currently if I look at `ThreadPool` implementation, I am not sure what is inside `threadId: String`.).

I don't mind having a bigger diff as long as we fix it the right way.

Btw, introducing the process id fix the issue right?

Reply:
got it! that make sense, @imback82 , thank you very much for the suggestions, i will update PR to fix it.

Reply:
hey @imback82 , update the code according your suggestion, please help review when you are available, thank you!

Reply:
hey, @imback82 , the build check failed  with below error:

Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project microsoft-spark-2-3_2.11: Execution default-compile of goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile failed: Plugin org.apache.maven.plugins:maven-compiler-plugin:3.1 or one of its dependencies could not be resolved: Could not transfer artifact org.apache.maven:maven-core:jar:2.0.9 from/to central (https://repo.maven.apache.org/maven2): Transfer failed for https://repo.maven.apache.org/maven2/org/apache/maven/maven-core/2.0.9/maven-core-2.0.9.jar: Connection reset -> [Help 1]
[ERROR] 

this is the log link:
https://dev.azure.com/dnceng/public/_build/results?buildId=969394&view=logs&j=bd1f0459-543a-55bf-0f8d-4369022127e8&t=a7ce2e4f-2245-52a5-b8ab-5e70fd44d0c0&l=17286

could you help check and solve the issue?

Reply:
ping @imback82